### PR TITLE
Add multiline handling functionality

### PR DESCRIPTION
**Pull request by jpike88**
Allows multiline logs to have each line styled properly.

### CODE REVIEW

**Comment by DABH** [comment]
Thanks for the PR!  Seems like some basic functionality has been broken though as tests are failing.  Can you please investigate and see what changes need to be made in order to get all the tests passing again?  Once everything passes, I'm happy to take a closer look and see if we can merge this.

File: lib/colors.js [hunk 1]
```diff
@@ -107,12 +107,26 @@ function applyStyle() {
...
   var i = nestedStyles.length;
+
+  var rebuiltString = '';
+
   while (i--) {
     var code = ansiStyles[nestedStyles[i]];
-    str = code.open + str.replace(code.closeRe, code.open) + code.close;
+
+    // multiline handling functionality
+    var lines = str.split(/[\r\n]+/)
```
Comment (DABH) [inline lib/colors.js]: Could this not be done with a one-line replace all, i.e. replace all `'\n'` with `code.close + '\n' + code.open` ?

**Comment by jpike88** [comment]
@DABH good to go

**Comment by DABH** [comment]
Thanks for updating.  This is looking pretty good now.  My only concern is, how much does this affect performance for the most common use case (single-line string)?  And is there any way we can mitigate any such performance risks?  First thought was sticking an if(str.indexOf('\n')) in front of the new line but that isn't particularly fast.  It's also possible that there is no silver bullet here and that we simply have to pay this price to make multiline strings work -- if so, ok, but I want to think a bit more (feel free to research/comment as well).

Also worth reading https://stackoverflow.com/questions/9750338/dynamic-vs-inline-regexp-performance-in-javascript and https://jsperf.com/regexp-indexof-perf/24 , there might be a more efficient approach than constructing that new RegExp every single time (that worries me a bit).

**Comment by jpike88** [comment]
I could make this an option and disable it by default.

**Comment by jpike88** [comment]
@DABH thoughts?

**Comment by DABH** [comment]
I'll take a look, give me a day or two ;)

**Comment by jpike88** [comment]
https://gist.github.com/jpike88/868b7912354907300ec790e34104557d

**Comment by jpike88** [comment]
My PR gives me 636ms,

Where the original code says 52ms. Definitely a difference there.

**Comment by DABH** [comment]
Nice, thanks for looking into this.  Ha, very glad we checked performance, yikes.

I'm just about ready to go here.  The only thing I'd ask is if you could add some test cases?  I'm thinking:
  - string with newlines and one style
  - string with newlines and multiple nested styles
If everything works as expected with those tests, then I'm happy to merge this and publish a new release.  Thanks again for your work on this!

**Comment by jpike88** [comment]
done, no worries!

**Pull request merged**